Sources/Core/Utilities/Money/Money+Currency.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  Money.swift
+//  PovioKit
+//
+//  Created by Marko Mijatovic on 04/06/2022.
+//  Copyright © 2022 Povio Labs. All rights reserved.
+//
+
+import Foundation
+
+public protocol Currency {
+  var code: CurrencyCode { get }
+  var symbol: String { get }
+}
+
+public extension Currency {
+  var currencyCode: String {
+    code.stringValue
+  }
+}
+
+public enum CurrencyCode: Codable {
```

Comment:
@borut-t 
Actually yes, this is much cleaner 
My approach was complicated because I started implementing it without enum, I was passing String to the generator to get correct Currency. At first, I was thinking to leave it open for possible variations (like "en_US", "usd", "USD", etc), but that would be to complicated so I switched to the enum but did not notice that generator is not needed any more...
Thanks, I will change it